### PR TITLE
Relocate class files from /include to /src/

### PR DESCRIPTION
**Pull request by zeroadam**
dbm, Diaspora, dfrn, and NotificationsManager moved to namespace. Includes and references in files updated.

### CODE REVIEW

**Comment by zeroadam** [comment]
Issue #3878

**Comment by MrPetovan** [comment]
Have you tried this code on a test instance for glaring 500 errors?

**Comment by zeroadam** [comment]
Still working on it, will likely be changes to come...

**Comment by MrPetovan** [comment]
I’m glad to have a fellow contributor not in the GMT+1/2 timezone!

**Comment by annando** [comment]
Is your Friendica installation still working? I think that especially ```dbm::is_result``` is called very often and not only in the places where you made the changes.

**Comment by zeroadam** [comment]
@annando Yes, I'm noticing many dbm:: in files that didn't have an include. I'm still on the search!

@MrPetovan haha! I know the feeling!

**Comment by annando** [comment]
@zeroadam okay, great that you are still on it. Don't forget the addons, there are dbm-calls there as well.

@MrPetovan concerning the class names, wouldn't DFRN a better name since it is an abbreviation? Dfrn looks strange in my eyes. Sames goes with DBM vs. Dbm.

BTW: Since I couldn't really fall asleep, a GMT+1 guy just crashed your GMT-x party :-)

**Comment by MrPetovan** [comment]
> @MrPetovan concerning the class names, wouldn't DFRN a better name since it is an abbreviation? Dfrn looks strange in my eyes. Sames goes with DBM vs. Dbm.

I personally would prefer those names as well, but I'm not sure if they match the expected case for classes in PSR-2. But since we already are two leaning in favor of capitalizing acronyms, we might as well go for it. @zeroadam, can you rename Dfrn to DFRN and Dbm to DBM? Thanks!

**Comment by zeroadam** [comment]
Yes, I figured you'd agree with @annando, so I started using caps in my changes!

**Comment by MrPetovan** [comment]
Well I for one am glad I have no secrets for you!

**Comment by MrPetovan** [comment]
As a general rule, please try to limit your pull requests to less than 100 modified files, as it makes the GitHub Web Interface pretty unresponsive and makes reviewing large PRs a bigger task than it already is. You can make multiple branches to that effect. It's too late for this PR, but please take it into account the next time you're taking a site-wide string replacement task like this one.

**Comment by zeroadam** [comment]
Ahh I see, I didn't think about that. Will remember for the future!

**Comment by zeroadam** [comment]
So after my latest changes, which were things I learned about by watching the httpd error log. Running my node on my issue branch seems to be functioning, i.e. no more Class not found errors.

Two strange things I am still seeing that I can't figure out are:

PHP Warning:  A non-numeric value encountered in /srv/http/friendica/addon/gravatar/lang/ru/strings.php on line 5

and

PHP Warning:  dns_get_record(): A temporary server error occurred. in /srv/http/friendica/include/network.php on line 481

File: doc/Developer-How-To-Move-Classes-to-src.md [hunk 1]
```diff
@@ -82,7 +82,7 @@ namespace Friendica\Core;
 
 use \dbm;
```
Comment (MrPetovan) [inline doc/Developer-How-To-Move-Classes-to-src.md]: This should be changed as well to `use Friendica\Database\DBM`.

File: doc/autoloader.md [hunk 2]
```diff
@@ -104,7 +104,7 @@ class Dfrn {
 
 mail_post($a){
 	...
-	Friendica\dfrn::mail($item, $owner);
+	Friendica\DFRN::mail($item, $owner);
```
Comment (MrPetovan) [inline doc/autoloader.md]: This should be changed to `Friendica\Protocol\DFRN`.

**Comment by MrPetovan** [comment]
> PHP Warning: dns_get_record(): A temporary server error occurred. in /srv/http/friendica/include/network.php on line 481

This will randomly happen because we query many different domains that aren't always available.

**Comment by annando** [comment]
@zeroadam For a **future** PR you could have a look at all the class files that are now used without a namespace. Several of them should be able to be relocated as well. (I'm thinking of xml.php and cache.php for example)

You could have a look at the files in the include folder if there are more static classes. The dba.php is sadly an exception since it currently contains some old style functions. Possibly these could be separated.

File: src/Protocol/DFRN.php [hunk 3]
```diff
@@ -1313,19 +1359,19 @@ public static function deliver($owner,$contact,$atom, $dissolve = false) {
...
+		if (strpos($xml, '<?xml') === false) {
 			logger('dfrn_deliver: phase 2: no valid XML returned');
 			logger('dfrn_deliver: phase 2: returned XML: ' . $xml, LOGGER_DATA);
 			return 3;
 		}
 
 		if ($contact['term-date'] > NULL_DATE) {
 			logger("dfrn_deliver: $url back from the dead - removing mark for death");
-			require_once('include/Contact.php');
+			include_once 'include/Contact.php';
```
Comment (MrPetovan) [inline src/Protocol/DFRN.php]: Why this change?

File: src/Protocol/DFRN.php [hunk 4]
```diff
@@ -1585,19 +1644,34 @@ private static function fetchauthor($xpath, $context, $importer, $element, $only
...
+					dbesc($contact["addr"]),
+					dbesc($contact["keywords"]),
+					dbesc($contact["bdyear"]),
+					dbesc($contact["bd"]),
+					intval($contact["hidden"]),
+					dbesc($contact["xmpp"]),
+					dbesc(DBM::date($contact["name-date"])),
+					dbesc(DBM::date($contact["uri-date"])),
+					intval($contact["id"]),
+					dbesc($contact["network"])
```
Comment (MrPetovan) [inline src/Protocol/DFRN.php]: As crazy as it is, this formatting made sense because the parameters lines followed the query lines.

A better way to improve readability would be to use `dba::update()`

File: src/Protocol/DFRN.php [hunk 5]
```diff
@@ -1831,7 +1913,8 @@ private static function process_suggestion($xpath, $suggestion, $importer) {
 			intval(0)
 		);
 
-		notification(array(
+		notification(
```
Comment (MrPetovan) [inline src/Protocol/DFRN.php]: I don’t see a problem with the former indentation, please revert it.

File: src/Protocol/DFRN.php (hunk 3 above)
Comment (zeroadam) [reply src/Protocol/DFRN.php]: this was a suggestion from phpcs, it first mentioned that require_once is a statement not a function. After removing the parenthesis, it then gave the suggestion that since it was a conditional include to use include_once instead of require_once.

File: src/Protocol/DFRN.php (hunk 4 above)
Comment (zeroadam) [reply src/Protocol/DFRN.php]: This was again just done based on suggestion from the tool. Something about calls being on their own line.

File: src/Protocol/DFRN.php (hunk 5 above)
Comment (zeroadam) [reply src/Protocol/DFRN.php]: Another suggestion from the tool, supposedly PRS2 something about the opening paren being the last symbol on the line. I will change it back

**Comment by annando** [comment]
If this PR not only contains the renamed classes but also adoptions to coding standards of every file then it will grow into a really huge monster.

**Comment by zeroadam** [comment]
You are quite right. I will refrain from making further standards changes
in this PR.

**Comment by MrPetovan** [comment]
Is your node still running this code and not failing spectacularly?

**Comment by zeroadam** [comment]
Yes, that's correct, no spectacular fails or error log entries.

**Comment by annando** [comment]
Some unspectacular ones? :-)

@MrPetovan If you want accept the PR now, you are free to do so. I'm too busy these days, currently I'm only having some minutes a day.

**Comment by MrPetovan** [comment]
Yeah, let's do it.
